src/common/ui-components/Map/Map.tsx

```diff
@@ -112,47 +132,118 @@ const Map: React.FC<MapProps> = ({
+          coordinates: [
+            feature.geometry.coordinates[0],
+            feature.geometry.coordinates[1]
+          ]
+        }
+      };
+    });
+  // get map bounds
+  const getBounds: () => BBox | null = function() {
+    const current = mapRef?.current;
+    if (!current) return null;
+    return current
+      .getMap()
+      .getBounds()
+      .toArray()
+      .flat() as BBox;
+  };
+  const bounds = getBounds();
+  const { clusters } = useSupercluster<GeoJsonProperties, ClusterProperties>({
+    points,
+    bounds,
+    zoom: viewPort.zoom,
+    options: { radius: clusteringRadius, maxZoom: viewPort.maxZoom }
+  });
 
-  const onViewportChange = (viewPort: ViewportProps) => {
+  const onViewportChange = (viewPort: ViewportState) => {
     const {
       width,
       height,
       transitionInterpolator,
       transitionDuration,
-      ...rest
-    } = viewPort;
-    setViewPort({
-      width: '100%',
-      height: '100%',
-      transitionInterpolator: undefined,
-      transitionDuration: undefined,
-      ...rest
-    });
+      latitude,
+      longitude,
+      zoom,
+      minZoom,
+      maxZoom
+    } = viewPort as ViewportProps;
+    setViewPort(viewPort);
   };
 
   return (
     <MapGL
-      className={className}
       {...viewPort}
       mapStyle={getMapStyle()}
+      width={'100%'}
+      height={'100vh'}
+      className={className}
+      ref={mapRef}
       onViewportChange={onViewportChange}
     >
-      {features.map((feature: Feature, id: number) => renderPin(feature, id))}
       <div className={styles.mapControls}>
         <GeolocateControl
           positionOptions={{ enableHighAccuracy: true }}
           trackUserLocation={true}
+          onViewportChange={() => {
+            /* NOOP, disables flying to location */
+          }}
           label={t('map.geolocate')}
         />
-        <div className={styles.mapControlsDivider} />
+        <div className={styles.mapControlsDivider}></div>
         <NavigationControl
           zoomInLabel={t('map.zoom_in')}
           zoomOutLabel={t('map.zoom_out')}
           showCompass={false}
         />
       </div>
+      {clusters.map(cluster => {
+        const [longitude, latitude] = cluster.geometry.coordinates;
+        const { cluster: isCluster } = cluster.properties;
+        // this is temporary, new designs should come soon
+        if (isCluster) {
+          const {
+            point_count: pointCount
+          } = cluster.properties as ClusterProperties;
+          return (
+            <Marker
+              key={`cluster-${cluster.id}`}
+              latitude={latitude}
+              longitude={longitude}
+            >
+              <div
+                className={styles.clusterMarker}
+                style={{
+                  width: `${10 + (pointCount / points.length) * 20}px`,
+                  height: `${10 + (pointCount / points.length) * 20}px`
+                }}
+              >
+                {pointCount}
```

Comment:
It looks like there's something funny going on with the clusters when two clusters are close together? Seems to be caused by numbers with more than one digit getting wrapped.

Reply:
Can you record it in a gif? Cannot reproduce it.